### ISSUE TITLE
Fail Case: robust ap fit fails if it doesn't select enough points

### ISSUE THREAD

**Issue by TomDonoghue** [fooof-tools/fooof#153]
In `_robust_ap_fit` a percentile threshold is used to sub-select points from a simple aperiodic fit, and re-fit these points. This percentile value is quite low. 

There is a fail-case, in which this percentile threshold leads to not enough point being selected, and the subsequent call to curve_fit fails. 

Notes:
- This seems to happen in pretty rare outlier cases, including cases in which there are already a small number of points, and (in at least one example) where there is a big 'dip' on the power spectrum.

Options:
- We could catch this error and just consider it a failed fit (clean failure mode)
- We could catch this error, and update the threshold if it happens, to try and get it to fit
- And/or, we could consider raising the threshold, to make this less likely

**Comment by voytek** [comment]
Ah yeah, great catch. The "cleanest" as you say, is to just catch and throw a failed-fit error.

Another option would be to, if it fails, instead of fitting the lowest _x_% of points, we fit the lowest _n_ points.

**Comment by TomDonoghue** [comment]
The only cases I've seen this happen on is some pretty funky spectra, for which I'm not sure it's worth tweaking just for the sake of getting some kind of fit. 

So I went with option 1, and just made it a clean failure mode. Now that I know what it is, the expert user can always manually update `_ap_percentile_thresh` if they want to push towards having this not happen. 

Fixed, and added to #152
